### PR TITLE
introduce `ilab` profile set doc

### PR DESCRIPTION
**Pull request by cdoern**

### CODE REVIEW

**Review by khaledsulayman — APPROVED** [review]

File: docs/cli/ilab-profile-set.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# `ilab profile set`
+
+### Usecase
+
+1. User needs to set:
+    a. Model Path for SDG, Eval, Serving, Training
+    b. GPUs for SDG, Eval, Serving, Training
+    c. Training Config Per-GPU (based on vram)
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:8]:
```suggestion
    c. Training Config Per-GPU (based on vRAM)
```

File: docs/cli/ilab-profile-set.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# `ilab profile set`
+
+### Usecase
+
+1. User needs to set:
+    a. Model Path for SDG, Eval, Serving, Training
+    b. GPUs for SDG, Eval, Serving, Training
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:7]: Serving just for vLLM, or vLLM and Llama CPP?

File: docs/cli/ilab-profile-set.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    c. Training Config Per-GPU (based on vram)
+2. User wants us to detect all of the above for them and provide a sane config
+
+### Workflow
+
+The user will run `ilab profile set`
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:13]: Will this be required before running other `ilab` commands? If not, what is the path for a user that does not do this?

File: docs/cli/ilab-profile-set.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+Alongside the various model paths and GPU amounts, this command will
+set the train profile for the following scenarions:
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:15]:
```suggestion
set the train profile for the following scenarios:
```

File: docs/cli/ilab-profile-set.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    1. CPU
+    2. single consumer GPU
+    3. multi consumer GPU
+    4. single server GPU
+    5. multi server GPU
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:21]: Are terms like `server` and `consumer` familiar to users? I myself don't fully understand what this means.

File: docs/cli/ilab-profile-set.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    6. MacOS (once MPS support is added)
+    
+There is also a Choose for me option which reads the Nvidia cards on the system and assigns a cfg+train profile based off the amount of vRAM
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:24]:
```suggestion
There is also a "Choose for me" option which reads the Nvidia cards on the system and assigns a cfg+train profile based off the amount of vRAM
```
What if I don't have a Nvidia card?

File: docs/cli/ilab-profile-set.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+ilab profile set -i
+How many Dedicated GPUs do you have?: 8
+Please choose a system profile to use
+[0] Choose for me (Hardware defaults)
+[1] CPU
+[2] Single Consumer GPU
+[3] Single Server GPU
+[4] Multi Consumer GPU
+[5] Multi Server GPU
+[6] MacOS
```
Comment (nathan-weinberg) [inline docs/cli/ilab-profile-set.md:38]: Is the assumption here systems 0-5 are Linux?

**Comment by nathan-weinberg** [comment]
@cdoern what is going on with this?

**Pull request closed without merging**